### ISSUE TITLE
Easier reflection padding implementation

### ISSUE THREAD

**Issue by unixpickle** [unixpickle/cnn-toys#2]
Write something like tf.layers.conv2d that supports padding='reflect'.
